### PR TITLE
Test codegen basic cases

### LINKED ISSUE #88: Unit test bundle for codegen basic cases
Tests should go in `codegen/codegen_test.go`. The input program should be as minimal as possible. There may be multiple test cases per bullet point:

- [x] scratch
- [x] http
- [x] git
- [x] env
- [x] dir
- [x] user
- [x] mkdir
- [x] mkfile
- [x] rm
- [x] copy

### PR DESCRIPTION
**Pull request by antonio-osorio**
Resolves #88

### CODE REVIEW

File: codegen/codegen_test.go [hunk 1]
```diff
@@ -53,6 +54,86 @@ func TestCodeGen(t *testing.T) {
...
+		"basic env",
+		[]string{"default"},
+		`
+		fs default() {
+			scratch
+			env "TEST_VAR" "test value"
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.Scratch().AddEnv("TEST_VAR", "test value"))
```
Comment (antonio-osorio) [inline codegen/codegen_test.go:135]: @hinshun Heads up that this is actually a no-op. The environment is taken into consideration when displaying the Tree.

Comment (hinshun) [reply codegen/codegen_test.go:135]: Yeah, that's fine. I'm okay with this.

**Comment by hinshun** [comment]
@antonio-osorio Are you still working on this? Are you planning to split off the rest of the items of #88 in a different PR?

**Comment by antonio-osorio** [comment]
@hinshun I have been using the work-in-progress label for the issues that I would like some feedback on (hence creating the PR) but not yet completed.

This is ready for merge now. But please note that the `env`, `dir` and `user` tests are no ops since they are not reflected in the Tree representation.

**Comment by hinshun** [comment]
@antonio-osorio If you have an empty `run ""` statement after, it'll inherit all those properties and be reflected in the Tree representation.

I think we can use the "Draft PR" feature, it seems to be what you're looking for, and I can still review them.

**Comment by antonio-osorio** [comment]
> I think we can use the "Draft PR" feature, it seems to be what you're looking for, and I can still review them.

Had not seen it before. I have converted the PR to a draft

**Comment by antonio-osorio** [comment]
> @antonio-osorio If you have an empty `run ""` statement after, it'll inherit all those properties and be reflected in the Tree representation.

`run ""` is not a valid statement, added an innocuous command.

**Pull request merged**